src/main/kotlin/edu/cornell/cs/apl/viaduct/selection/ProtocolSelector.kt

```diff
@@ -0,0 +1,11 @@
+package edu.cornell.cs.apl.viaduct.selection
+
+import edu.cornell.cs.apl.viaduct.syntax.Protocol
+import edu.cornell.cs.apl.viaduct.syntax.Variable
+import edu.cornell.cs.apl.viaduct.syntax.intermediate.DeclarationNode
+import edu.cornell.cs.apl.viaduct.syntax.intermediate.LetNode
+
+interface ProtocolSelector {
+    fun selectLet(assignment: Map<Variable, Protocol>, node: LetNode): Set<Protocol>
```

Comment:
It would also be useful to have comments explaining what this function is supposed to do. It is especially important to explain the `assignment` argument since it's a current/partial assignment.